plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/generator/SQLGeneratorSelectManyFromData.java

```diff
@@ -24,48 +25,40 @@
 import org.jkiss.dbeaver.ui.controls.resultset.IResultSetController;
 import org.jkiss.dbeaver.ui.controls.resultset.ResultSetRow;
 
-import java.util.Collection;
 import java.util.List;
 
 public class SQLGeneratorSelectManyFromData extends SQLGeneratorResultSet {
 
     @Override
-    public void generateSQL(DBRProgressMonitor monitor, StringBuilder sql, IResultSetController object) {
-        Collection<DBDAttributeBinding> keyAttributes = getKeyAttributes(monitor, object);
+    public void generateSQL(DBRProgressMonitor monitor, StringBuilder sql, IResultSetController object) throws DBException {
+        List<DBDAttributeBinding> whereAttr = getSelectedAttributes();
+        if (whereAttr.size() > 1) {
```

Comment:
It is supposed to work with multiple attributes as well. In that case, try to put them together with an AND statement

Reply:
@HocKu7 Thanks for reviewing, it's updated.